stm32-modules/heater-shaker/include/heater-shaker/gcodes.hpp

```diff
@@ -676,6 +675,116 @@ struct SetSerialNumber {
+    ** Example: M998.D 0 selects WHITE_ON mode and turns the white LED on
+    */
+    using ParseResult = std::optional<SetLEDDebug>;
+    static constexpr auto prefix = std::array{'M', '9', '9', '8', '.', 'D', ' '};
+    static constexpr const char* response = "M998.D OK\n";
+    LED_MODE mode;
+
+    template <typename InputIt, typename InputLimit>
+    requires std::forward_iterator<InputIt> &&
+        std::sized_sentinel_for<InputLimit, InputIt>
+    static auto write_response_into(InputIt buf, InputLimit limit) -> InputIt {
+        return write_string_to_iterpair(buf, limit, response);
+    }
+
+    template <typename InputIt, typename Limit>
+    requires std::forward_iterator<InputIt> &&
+        std::sized_sentinel_for<Limit, InputIt>
+    static auto parse(const InputIt& input, Limit limit)
+        -> std::pair<ParseResult, InputIt> {
+        auto working = prefix_matches(input, limit, prefix);
+        if (working == input) {
+            return std::make_pair(ParseResult(), input);
+        }
+
+        auto value_res = parse_value<uint8_t>(working, limit);
+
+        if (!value_res.first.has_value()) {
+            return std::make_pair(ParseResult(), input);
+        }
+        LED_MODE tempMode = static_cast<LED_MODE>(value_res.first.value());
+        return std::make_pair(ParseResult(SetLEDDebug{.mode = tempMode}),
+                              value_res.second);
+    }
+};
+
+struct IdentifyModuleStartLED {
+    /*
+    ** IdentifyModuleStartLED uses a random gcode, M998, adjacent to the firmware
+    ** update gcode, M997, to start blinking the white LED
+    ** Format: M998
+    ** Example: M998 starts blinking the white LED
+    */
+    using ParseResult = std::optional<IdentifyModuleStartLED>;
+    static constexpr auto prefix = std::array{'M', '9', '9', '8'};
+    static constexpr const char* response = "M998 OK\n";
+    LED_MODE mode;
+
+    template <typename InputIt, typename InputLimit>
+    requires std::forward_iterator<InputIt> &&
+        std::sized_sentinel_for<InputLimit, InputIt>
+    static auto write_response_into(InputIt buf, InputLimit limit) -> InputIt {
+        return write_string_to_iterpair(buf, limit, response);
+    }
+
+    template <typename InputIt, typename Limit>
+    requires std::forward_iterator<InputIt> &&
+        std::sized_sentinel_for<Limit, InputIt>
+    static auto parse(const InputIt& input, Limit limit)
+        -> std::pair<ParseResult, InputIt> {
+        auto working = prefix_matches(input, limit, prefix);
+        if (working == input) {
+            return std::make_pair(ParseResult(), input);
+        }
+        if (working != limit && !std::isspace(*working)) {
+            return std::make_pair(ParseResult(), input);
+        }
+        return std::make_pair(ParseResult(IdentifyModuleStartLED()), working);
+    }
+};
+
+struct IdentifyModuleStopLED {
+    /*
+    ** IdentifyModuleStopLED uses a random gcode, M999, adjacent to the identify
```

Comment:
[nit] `M999` is an emergency stop handling gcode in a lot of other implementations. can we pick something else?